Improve contrast when deleting terms in list table via AJAX

nikunj8866 opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/47101

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

Commit:
Improved contrast when deleting terms via ajax

nikunj8866 commented:
> The following accounts have interacted with this PR and/or linked issues. I will continue to update these lists as activity occurs. You can also manually ask me to refresh this list by adding the `props-bot` label.
> 
> ## Unlinked Accounts
> The following contributors have not linked their GitHub and WordPress.org accounts: @[nhatkar@updraftplus.com](mailto:nhatkar@updraftplus.com).
> 
> Contributors, please [read how to link your accounts](https://make.wordpress.org/core/2020/03/19/associating-github-accounts-with-wordpress-org-profiles/) to ensure your work is properly credited in WordPress releases.
> 
> **To understand the WordPress project's expectations around crediting contributors, please [review the Contributor Attribution page in the Core Handbook](https://make.wordpress.org/core/handbook/best-practices/contributor-attribution-props/).**

Please use @nikunj8866 WordPress profile.

nikunj8866 commented:
### After Patch:

![image](https://github.com/user-attachments/assets/32b9421f-e9d3-42cd-bdbe-e736ce2db979)

nikunj8866 committed:
Merge branch 'WordPress:trunk' into 47101-improve-contrast-when-deleting-terms-in-list-table-via-ajax

nikunj8866 committed:
Disabled interactive elements in the row during deletion process

nikunj8866 committed:
Fixed Javascript coding standard errors

SirLouen commented on src/js/_enqueues/admin/tags.js:
```diff
@@ -31,6 +31,12 @@ jQuery( function($) {
 		if ( r ) {
 			data = t.attr('href').replace(/[^?]*\?/, '').replace(/action=delete/, 'action=delete-tag');
 
+			tr.children().css('backgroundColor', '#faafaa');
```
The 3 blocks basically do the same.  You can create a function with some parameters, and use such function for the 3, saving much repeition.

nikunj8866 committed:
Optimized code

joedolson committed:
Add a spoken message after delete action, set focus after deletion

When the currently focused control is deleted, a loss of focus is triggered; need to explicitly set the focus. Standard practice is the next item in the collection if there is one. If not, I set it to the previous; if neither, to the tag input.

Also adds wp.a11y.speak after each action.

joedolson commented:
I added some changes to improve the accessibility of these interactions. Notes are in the commit message.
